command/rootchain/helper/rootchain_interactor.go

```diff
@@ -0,0 +1,156 @@
+package helper
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/jsonrpc"
+	"github.com/umbracle/ethgo/wallet"
+)
+
+type RootchainInteractor interface {
+	Call(from types.Address, to types.Address, input []byte) (string, error)
+	SendTransaction(transaction *ethgo.Transaction, signer ethgo.Key) (*ethgo.Receipt, error)
+	GetPendingNonce(address types.Address) (uint64, error)
+	ExistsCode(contractAddr types.Address) (bool, error)
+	FundAccount(account types.Address) (types.Hash, error)
+}
+
+var _ RootchainInteractor = (*DefaultRootchainInteractor)(nil)
+
+type DefaultRootchainInteractor struct {
+	provider *jsonrpc.Client
+}
+
+func NewDefaultRootchainInteractor(ipAddress string) (*DefaultRootchainInteractor, error) {
+	provider, err := jsonrpc.NewClient(ipAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	return &DefaultRootchainInteractor{provider: provider}, nil
+}
+
+func (d *DefaultRootchainInteractor) Call(from types.Address, to types.Address, input []byte) (string, error) {
+	toAddr := ethgo.Address(to)
+	callMsg := &ethgo.CallMsg{
+		From:     ethgo.Address(from),
+		To:       &toAddr,
+		Data:     input,
+		GasPrice: defaultGasPrice,
+		Gas:      big.NewInt(defaultGasLimit),
+	}
+
+	return d.provider.Eth().Call(callMsg, ethgo.Pending)
+}
+
+func (d *DefaultRootchainInteractor) SendTransaction(txn *ethgo.Transaction,
+	privKey ethgo.Key) (*ethgo.Receipt, error) {
+	if txn.GasPrice == 0 {
+		txn.GasPrice = defaultGasPrice
+	}
+
+	if txn.Gas == 0 {
+		txn.Gas = defaultGasLimit
+	}
+
+	chainID, err := d.provider.Eth().ChainID()
+	if err != nil {
+		return nil, err
+	}
+
+	signer := wallet.NewEIP155Signer(chainID.Uint64())
+	if txn, err = signer.SignTx(txn, privKey); err != nil {
+		return nil, err
+	}
+
+	data, err := txn.MarshalRLPTo(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	txnHash, err := d.provider.Eth().SendRawTransaction(data)
+	if err != nil {
+		return nil, err
+	}
+
+	receipt, err := d.waitForReceipt(txnHash)
```

Comment:
maybe just `return d.waitForReceipt(txnHash)` ?

Reply:
`DefaultRootchainInteractor` is gone in favor of `TxRelayer`, but applied feedback :wink: https://github.com/0xPolygon/polygon-edge/pull/989/commits/3bd627d9aa808d76e850cb9575e7d1a38d4d5007#diff-c83c8a1e714409dd49d38a1ac6f69af3a22eeee9947090430b37619db121ccf7R93